# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundParentImpl.cpp
# Commit: 8132dd939023
# Full Hash: 8132dd9390233328c8da7072f36859424a2b079d
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-10-14 21:42:48
# Regressor Bug: 1657404
# File Overlap Count: 2
# Description:
#   Bug 1657404 - Change PGamepadEventChannel to "refcounted protocol" r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93119
# ==============================================================================

diff -r 186ca9e93e1d -r 8132dd939023 ipc/glue/BackgroundParentImpl.cpp
--- a/ipc/glue/BackgroundParentImpl.cpp	Wed Oct 07 11:34:02 2020 +0000
+++ b/ipc/glue/BackgroundParentImpl.cpp	Wed Oct 14 15:58:36 2020 +0000
@@ -1135,30 +1135,9 @@
 }
 
 // Gamepad API Background IPC
-dom::PGamepadEventChannelParent*
+already_AddRefed<dom::PGamepadEventChannelParent>
 BackgroundParentImpl::AllocPGamepadEventChannelParent() {
-  RefPtr<dom::GamepadEventChannelParent> parent =
-      new dom::GamepadEventChannelParent();
-
-  return parent.forget().take();
-}
-
-bool BackgroundParentImpl::DeallocPGamepadEventChannelParent(
-    dom::PGamepadEventChannelParent* aActor) {
-  MOZ_ASSERT(aActor);
-  RefPtr<dom::GamepadEventChannelParent> parent =
-      dont_AddRef(static_cast<dom::GamepadEventChannelParent*>(aActor));
-  return true;
-}
-
-mozilla::ipc::IPCResult
-BackgroundParentImpl::RecvPGamepadEventChannelConstructor(
-    PGamepadEventChannelParent* aActor) {
-  MOZ_ASSERT(aActor);
-  if (!static_cast<dom::GamepadEventChannelParent*>(aActor)->Init()) {
-    return IPC_FAIL_NO_REASON(this);
-  }
-  return IPC_OK();
+  return dom::GamepadEventChannelParent::Create();
 }
 
 dom::PGamepadTestChannelParent*